File: sukjoo/Programmers/Level_2/Lv2_영어_끝말잇기.java
public class Lv2_영어_끝말잇기 {
    public int[] solution(int n, String[] words) {
        int[] answer = new int [2];
        
        for (int i = 0 ; i < words.length; i++){
            for(int j = 0 ; j < i ; j++){
                if (words[j].equals(words[i])&& i != j ){
                    answer[0] = (i % n) + 1;
                    answer[1] = (i / n) + 1;
                    return answer;
                }
            }
            
            if(i>0&&words[i-1].charAt(words[i-1].length()-1) != words[i].charAt(0)){
                answer[0] = (i % n) +1 ;
                answer[1] = (i / n) +1 ;
                break;
            }
            else {
                answer[0] = 0;
                answer[1] = 0;
            }
            
        }
        System.out.println();

        return answer;
    }
    
}

File: dohyeon/Programmers/Level_2/Lv2_영어_끝말잇기.py
def solution(n, words):
    answer = []
    stack = []
    user = {}
    words_index = 0
    for i in range(len(words)) :
        try :
            user[(i % n) + 1].append(words[i])
        except :
            user[(i % n) + 1] = [words[i]]
        if words[i] in stack :
            words_index = i
            break
        stack.append(words[i])
        if (len(stack) >= 2 and stack[-2][-1] != words[i][0]):
            words_index = i
            break
    if words_index != 0 :
        answer.append((words_index % n) + 1)
        answer.append(user[(words_index % n) + 1].index(words[words_index]) + 1)
    elif words_index == 0 :
        answer = [0, 0]
        
    return answer

print(solution(3, ["tank", "kick", "know", "wheel", "land", "dream", "mother", "robot", "tank"])) #	[3,3]
print(solution(5, ["hello", "observe", "effect", "take", "either", "recognize", "encourage", "ensure", "establish", "hang", "gather", "refer", "reference", "estimate", "executive"])) # [0,0]
print(solution(2, ["hello", "one", "even", "never", "now", "world", "draw"])) # [1,3]